test/compilable/test18295.d

```diff
@@ -0,0 +1,19 @@
+// REQUIRED_ARGS: -dip1000
+
+// https://issues.dlang.org/show_bug.cgi?id=18295
+
+// See test/fail_compilation/test18295.d for the non-`-dip1000` version`
+
+scope class C { int i; }    // Notice the use of `scope` here
```

Comment:
`scope class` is irrelevant for this test, please remove the misleading comment.
Only the fact that `c` is a scoped class reference matters for the return scope check.